chat-sdk-ui/src/main/java/co/chatsdk/ui/SendMultipleImagesAtOnceActivity.java

```diff
@@ -0,0 +1,181 @@
+package co.chatsdk.ui;
+
+import android.app.Activity;
+import android.content.ClipData;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.provider.MediaStore;
+import android.view.View;
+import android.widget.TextView;
+
+import androidx.annotation.Nullable;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.github.chrisbanes.photoview.PhotoView;
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
+import com.theartofdev.edmodo.cropper.CropImage;
+
+import java.io.File;
+import java.util.ArrayList;
+
+import co.chatsdk.ui.chat.MediaSelector;
+import co.chatsdk.ui.main.BaseActivity;
+import co.chatsdk.ui.utils.Cropper;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+
+import static co.chatsdk.ui.chat.MediaSelector.CHOOSE_PHOTO;
+import static com.theartofdev.edmodo.cropper.CropImage.CROP_IMAGE_ACTIVITY_REQUEST_CODE;
+
+public class SendMultipleImagesAtOnceActivity extends BaseActivity {
+
+    private String imageURI;
+    private ArrayList<Uri> uriArrayList;
```

Comment:
Why is this defined here too? We only need one array list and it's defined in the adapter as well.

Reply:
That is the Uri Array List that is brought into the activity if the user selects multiple images before entering this activity. It is added to the uriArrayList in the Adapter on line 80. That is also why that one has to be public.